src/apps/aave/positions.ts

```diff
@@ -111,211 +150,152 @@ const hook: PositionsHook = {
+    return [
+      reservesData.flatMap((reserveData, i) => {
+        const supplyApy = getApyFromRayApr(reserveData.liquidityRate)
+        const variableBorrowApy = getApyFromRayApr(
+          reserveData.variableBorrowRate,
         )
-      const variableDebtRewardsInfo =
-        userIncentives?.vTokenIncentivesUserData.userRewardsInformation.filter(
-          (info) => info.userUnclaimedRewards > 0n,
-        )
-      const stableDebtRewardsInfo =
-        userIncentives?.sTokenIncentivesUserData.userRewardsInformation.filter(
-          (info) => info.userUnclaimedRewards > 0n,
+        const stableBorrowApy = getApyFromRayApr(
+          userReserveData?.[i].stableBorrowRate || reserveData.stableBorrowRate,
         )
 
-      return [
-        // AToken
-        useAToken &&
-          ({
-            type: 'app-token-definition',
-            networkId,
-            address: reserveData.aTokenAddress.toLowerCase(),
-            tokens: [
-              { address: reserveData.underlyingAsset.toLowerCase(), networkId },
-            ],
-            availableShortcutIds: ['deposit', 'withdraw'],
-            displayProps: {
-              title: reserveData.symbol,
-              description: `Supplied (APY: ${supplyApy.toFixed(2)}%)`,
-              imageUrl: AAVE_LOGO,
-            },
-            dataProps: {
-              manageUrl,
-              termsUrl: AAVE_TERMS_URL,
-              contractCreatedAt:
-                AAVE_CONTRACT_CREATED_AT[
-                  getTokenId({
-                    networkId,
-                    address: reserveData.aTokenAddress.toLowerCase(),
-                  })
-                ],
-              tvl: toSerializedDecimalNumber(
-                toDecimalNumber(totalSupplies[i], lpTokenDecimals[i]),
-              ),
-              yieldRates: [
+        // Include a token if the user has a balance
+        // or if no user data is available (when querying all positions)
+        const useAToken =
+          !userReserveData || userReserveData[i].scaledATokenBalance > 0n
+        const useVariableDebt =
+          !userReserveData || userReserveData[i].scaledVariableDebt > 0n
+        const useStableDebt =
+          !userReserveData || userReserveData[i].principalStableDebt > 0n
+
+        return [
+          // AToken
+          useAToken &&
+            ({
+              type: 'app-token-definition',
+              networkId,
+              address: reserveData.aTokenAddress.toLowerCase(),
+              tokens: [
                 {
-                  percentage: supplyApy,
-                  label: t('yieldRates.earningsApy'),
-                  tokenId: getTokenId({
-                    networkId,
-                    address: reserveData.underlyingAsset.toLowerCase(),
-                  }),
+                  address: reserveData.underlyingAsset.toLowerCase(),
+                  networkId,
                 },
               ],
-              earningItems: aTokenRewardsInfo?.length
```

Comment:
We could/should calculate the accrued rewards here (not yet claimed).

So maybe we should bite the bullet and copy the calculation logic? 😅 
Instead of using `getAllUserRewards`.